File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/JobPosting.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace HRManagementSystem.Domain.Entities;

public class JobPosting : BaseEntity
{
    [Required]
    [StringLength(200, MinimumLength = 5)]
    public required string Title { get; set; }
    [StringLength(2000)]
    public string? Description { get; set; }
    public string? Requirements { get; set; }
    public DateTime PostedDate { get; set; }
    public DateTime? ClosingDate { get; set; }
    public bool? IsActive { get; set; }

    // Navigation properties
    public int DepartmentId { get; set; }
    public Department? Department { get; set; }
    public int DesignationId { get; set; }
    public Designation? Designation { get; set; }

    public ICollection<JobApplication> JobApplications { get; set; } = new List<JobApplication>();
}

File: frontend/src/app/models/recruitment/job-posting.ts
export interface JobPosting {
  id: number;
  title: string;
  description?: string;
  requirements?: string;
  postedDate: string;
  closingDate?: string;
  isActive?: boolean;
  departmentId: number;
  designationId: number;
  departmentName?: string;
  designationName?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateJobPostingDto {
  title: string;
  description?: string;
  requirements?: string;
  postedDate: string;
  closingDate?: string;
  isActive?: boolean;
  departmentId: number;
  designationId: number;
}

export interface UpdateJobPostingDto extends CreateJobPostingDto {}
